app/models/media_resource.rb

```diff
@@ -43,8 +43,13 @@ def copy_media(force = false)
   end
 
   def media_url
+    media_url_for_base(episode.base_published_url) if episode
+  end
+
+  def media_url_for_base(base_published_url)
     ext = File.extname(original_url || '')
     ext = '.mp3' if ext.blank?
-    "#{episode.base_published_url}/#{guid}#{ext}" if episode
+    "#{base_published_url}/#{guid}#{ext}"
   end
+
```

Comment:
I'm not sure I see the practical change here, but it looks like this is covering a case where episodeA.media = mediaA and mediaA.episode != episodeA, am I getting that right?

Reply:
This was tracking down an N+1 query.  When rendering the `/api/v1/episodes` list, it was: 

1. SQL-ing the single page of episodes from the episode table
2. Calling this for each episode
3. The `if episode` and `base_published_url` were actually reloading every episode in the list, individually by `WHERE ID = ?`.

Reply:
(I'm curious if you have other ideas on how to prevent it reloading the episodes-by-id ... this was just the quick and dirty way for me ... "inject the episode")